From 1ff1906545bacb3eb7608155984dd4642647b736 Mon Sep 17 00:00:00 2001
From: akallabeth <akallabeth@posteo.net>
Date: Sat, 27 Feb 2021 08:29:46 +0100
Subject: [PATCH 01/36] Added compatibility define

---
 winpr/include/winpr/shell.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/winpr/include/winpr/shell.h b/winpr/include/winpr/shell.h
index 72931a01b..f88291e43 100644
--- a/winpr/include/winpr/shell.h
+++ b/winpr/include/winpr/shell.h
@@ -86,6 +86,7 @@ typedef enum
 
 /* Deprecated, here for compatibility */
 #define FD_SHOWPROGRESSUI FD_PROGRESSUI
+#define FD_WRITESTIME FD_WRITETIME
 
 /* FILEDESCRIPTOR.dwFileAttributes */
 #define FILE_ATTRIBUTE_READONLY 0x00000001
-- 
2.30.2

